source/_integrations/homekit_controller.markdown

```diff
@@ -45,67 +45,82 @@ ha_integration_type: integration
+There are different methods to add a HomeKit device to Home Assistant:
 
-## Supported devices
+- [via Ethernet or Wi-Fi](#adding-a-homekit-device-via-ethernet-or-wi-fi)
+- [via Bluetooth](#adding-a-homekit-device-through-bluetooth)
+- via Thread
+  - [by using Home Assistant’s preferred Thread network](#adding-a-homekit-device-to-a-thread-network-via-home-assistant)
+  - [by using Apple Thread border router](#adding-a-homekit-device-via-apple-thread-border-router)
 
-There is currently support for the following device types within Home Assistant:
+## Adding a HomeKit device via Ethernet or Wi-Fi
 
-- Alarm Control Panel (HomeKit security system)
-- Climate (HomeKit thermostats and air conditioners)
-- Cover (HomeKit garage door openers, windows, or window coverings)
-- Light (HomeKit lights)
-- Lock (HomeKit lock)
-- Switch (HomeKit switches, outlets and valves)
-- Binary Sensor (HomeKit motion, contact, occupancy, carbon monoxide and smoke sensors)
-- Sensor (HomeKit humidity, temperature, co2 and light level sensors)
-- Fan
-- Air Quality
-- Humidifier (HomeKit humidifiers and dehumidifiers)
-- Automation Triggers (HomeKit 'stateless' accessories like buttons, remotes and doorbells)
+The HomeKit controller integration automatically detects HomeKit [compatible devices](#supported-devices) on your network when they are ready to pair.
 
-<div class='note'>
+### Prerequisites
 
-  If your device is not on this list then you may still be able to pair it, and it will be visible in the Device Registry. But Home Assistant may not create entities for it.
+- If you do not have the [`default_config`](/integrations/default_config/) integration, add [`zeroconf`](/integrations/zeroconf/) to your configuration file.
+- Find your HomeKit pairing code. The code is on the device itself, or on the packaging. If your device has a screen, it may be shown on screen.
+   - There is no way to recover this if you do not have it. In this case, you will need to contact the manufacturer to see what options you have.
+- Make sure your device is powered up.
+- If your device is in your network but is paired with an Apple device via HomeKit, remove the device from the Apple Home app.
+   - Otherwise you won't be able to pair it with Home Assistant.
+   - HomeKit devices can only be paired to a single controller at once.
+- If your device is not already in your network, join the device to your network:
+   - Follow the instructions of the manufacturer on how to join the device to the network
+   - If you do not have the instructions, or the device is HomeKit only:
+     - Pair it with the Apple Home app (using the iOS Home app).
+     - After it has been paired with the Apple Home app, remove the device from the app again.
+     - Do not reset the device.
```

Comment:
It doesn't excite me completely. Another stab:

```suggestion
- Make sure the device is on your network, but not paired with another HomeKit controller. Depending on the device you need to go through a different set of steps:
  - If your device is not already in your network: **Join** the device to your network:
    - Follow the instructions of the manufacturer on how to join the device to the network
    - If the device is HomeKit only, simply pair it with the Apple Home app, then follow the next bullet point.
  - If your device is in your network but is paired with an Apple device via HomeKit: **Remove** the device from the Apple Home app.
    - Otherwise you won't be able to pair it with Home Assistant.
  - **Note:** Adding the device to the Home app and then removing it again has two effects:
    - It adds the device to your network. It stays in the network even after you removed it from the app.
    - Removing the device from the app opens it up for pairing with Home Assistant. HomeKit devices can only be paired to a single controller at once.
```